frontend/dockerfile/dockerfile2llb/convert.go

```diff
@@ -1324,8 +1324,19 @@ func prefixCommand(ds *dispatchState, str string, prefixPlatform bool, platform
 	if ds.stageName != "" {
 		out += ds.stageName + " "
 	}
+	var format string
+	switch  {
+	case ds.cmdTotal < 10:
+		format = "%d/%d]"
+	case ds.cmdTotal < 100:
+		format = "%2d/%d]"
+	case ds.cmdTotal < 1000:
+		format = "%3d/%d]"
+	default:
+		format = "%d/%d]"
+	}
```

Comment:
@tonistiigi updated, but took a slightly different approach; using log looked quite overcomplicated; as there's a lot of conversion, rounding, and twice a printf needed;

```go
	format := fmt.Sprintf("%%%dd/%%d] ",  int(math.Round(0.5+math.Log10(float64(ds.cmdTotal)))))
	out += fmt.Sprintf(format, ds.cmdIndex, ds.cmdTotal)
```

(even thinking if we need anything > 100 steps in a single stage in a dockerfile)


```patch
diff --git a/frontend/dockerfile/dockerfile2llb/convert.go b/frontend/dockerfile/dockerfile2llb/convert.go
index 4a41bc73..2fad0ab8 100644
--- a/frontend/dockerfile/dockerfile2llb/convert.go
+++ b/frontend/dockerfile/dockerfile2llb/convert.go
@@ -5,6 +5,7 @@ import (
        "context"
        "encoding/json"
        "fmt"
+       "math"
        "net/url"
        "path"
        "path/filepath"
@@ -1324,18 +1325,8 @@ func prefixCommand(ds *dispatchState, str string, prefixPlatform bool, platform
        if ds.stageName != "" {
                out += ds.stageName + " "
        }
-       var format string
-       switch  {
-       case ds.cmdTotal < 10:
-               format = "%d/%d]"
-       case ds.cmdTotal < 100:
-               format = "%2d/%d]"
-       case ds.cmdTotal < 1000:
-               format = "%3d/%d]"
-       default:
-               format = "%d/%d]"
-       }
        ds.cmdIndex++
+       format := fmt.Sprintf("%%%dd/%%d] ",  int(math.Round(0.5+math.Log10(float64(ds.cmdTotal)))))
        out += fmt.Sprintf(format, ds.cmdIndex, ds.cmdTotal)
        return out + str
 }
```

Reply:
You can do this one line with:

```
out += fmt.Sprintf("%[1]*d/%d] ",  int(1+math.Log10(float64(ds.cmdTotal))), ds.cmdIndex, ds.cmdTotal)
```

Reply:
I agree that `log` is overkill in here (and don't care about the >100 case), but do make the change that puts the length on the single `Sprintf` line instead of having the multiple versions of format definitions.

Reply:
Whoops, had this page open in a tab, so missed your update

> ```
> out += fmt.Sprintf("%*d/%d] ",  int(1+math.Log10(float64(ds.cmdTotal))), ds.cmdIndex, ds.cmdTotal)
> ```

oh! TIL about `%*..` didn't know that

Also wondering why I had the `math.Round()` (I know I had `1+` originally, but think I was testing some things.. silly 😅)